dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java

```diff
@@ -587,4 +501,59 @@ private static List<OrganisationUnit> getAccessibleDescendants(
   private static boolean isProgramAccessRestricted(Program program) {
     return program != null && (program.isClosed() || program.isProtected());
   }
+
+  private void mapDataElementFilters(EventSearchParams params, List<QueryItem> dataElementFilters) {
+    for (QueryItem item : dataElementFilters) {
+      for (QueryFilter filter : item.getFilters()) {
+        params.addDataElementFilter((DataElement) item.getItem(), filter);
+      }
+    }
+  }
+
+  private void mapAttributeFilters(EventSearchParams params, List<QueryItem> attributeFilters) {
+    for (QueryItem item : attributeFilters) {
+      if (item.getFilters().isEmpty()) {
```

Comment:
This difference in behavior between attributes and data elements is because we support filtering on attributes with no value, meaning to check if the attribute is present, but we do not support the same for data element?

Reply:
Yes 😄